homeassistant/components/websocket_api/automation.py

```diff
@@ -0,0 +1,149 @@
+"""Commands part of Websocket API."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from homeassistant.const import CONF_TARGET
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er, target as target_helpers
+from homeassistant.helpers.service import (
+    async_get_all_descriptions as async_get_all_service_descriptions,
+)
+from homeassistant.helpers.trigger import (
+    async_get_all_descriptions as async_get_all_trigger_descriptions,
+)
+from homeassistant.helpers.typing import ConfigType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(slots=True, kw_only=True)
+class _ComponentLookupData:
+    """Helper data structure for looking up automation components."""
+
+    component: str
+    domains: set[str]
+    device_classes: set[str]
+    supported_features: list[int]
+
+    def matches(self, entity_entry: er.RegistryEntry) -> bool:
+        """Return if entity matches the filters."""
+        if self.domains and entity_entry.domain not in self.domains:
+            return False
+        if self.device_classes:
+            entity_device_class = entity_entry.device_class
+            if (
+                entity_device_class is None
+                or entity_device_class not in self.device_classes
+            ):
+                return False
+        if self.supported_features:
+            entity_supported_features = entity_entry.supported_features
+            if entity_supported_features is None or not any(
+                feature & entity_supported_features
+                for feature in self.supported_features
+            ):
+                return False
+        return True
+
+
+def _build_component_lookup_data(
+    component: str, target_description: dict[str, Any]
+) -> _ComponentLookupData:
+    """Build automation component lookup data from target description."""
+    domains: set[str] = set()
+    device_classes: set[str] = set()
+    supported_features: list[int] = []
+
+    selector_entities = target_description.get("entity", [])
+    for selector_entity in selector_entities:
+        domains.update(selector_entity.get("domain", []))
+        device_classes.update(selector_entity.get("device_class", []))
+        supported_features.extend(selector_entity.get("supported_features", []))
+
+    return _ComponentLookupData(
+        component=component,
+        domains=domains,
+        device_classes=device_classes,
+        supported_features=supported_features,
+    )
+
+
+async def _async_get_components_for_target(
+    hass: HomeAssistant,
+    target_selector: ConfigType,
+    expand_group: bool,
+    descriptions: dict[str, dict[str, Any] | None],
```

Comment:
descriptions is not very descriptive. can we name it better? Maybe `automation_descriptions`?

Reply:
renamed to `component_descriptions` since it does not describe automations